client/src/context/useProfileContext.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useContext, createContext, FunctionComponent, useEffect, useCallback } from 'react';
+import { ProfileApiData } from '../interface/profile/ProfileApiData';
+import { SearchProfilesApiData, ProfilesApiDataSuccess } from '../interface/profile/Profiles';
+import { Profile } from '../interface/profile/Profile';
+import { getMyProfile } from '../helpers/APICalls/getProfilesApi';
+
+interface IProfileContext {
+  myProfile: Profile | null | undefined;
+  profiles: Profile[] | undefined;
+  updateMyProfileContext: (data: ProfileApiData) => void;
+  updateProfilesContext: (data: SearchProfilesApiData) => void;
+}
+
+export const ProfileContext = createContext<IProfileContext>({
+  myProfile: undefined,
+  profiles: [],
+  updateMyProfileContext: () => null,
+  updateProfilesContext: () => null,
+});
+
+export const ProfileProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [myProfile, setMyProfile] = useState<Profile | null | undefined>();
```

Comment:
Optional: I would highly recommend storing the logged in user's profile within the existing AuthContext and storing the list of profiles within the 'Listings' Page (we don't need a context for this). Overall, this context can be avoided :)

Reply:
I removed the profileContext and added the profile object in AuthContext.